zim/gui/pageview/__init__.py

```diff
@@ -1390,18 +1390,18 @@ def _get_close_anchor_or_object_id(self, iter):
 			return None
 
 	def _get_implict_anchor_if_heading(self, iter):
-		text = self._get_heading_text(iter)
+		text = self.get_heading_text(iter)
 		return heading_to_anchor(text) if text else None
 
-	def _get_heading_text(self, iter):
+	def get_heading_text(self, iter):
 		line_start = iter.copy() if iter.starts_line() else self.get_iter_at_line(iter.get_line())
 		is_heading = any(filter(_is_heading_tag, line_start.get_tags()))
 		if not is_heading:
 			return None
 
 		line_end = line_start.copy()
 		line_end.forward_line()
-		return line_start.get_text(line_end)
+		return line_start.get_text(line_end).strip()
```

Comment:
This is the breaking change, as confirmed by the issue getting resolved by taking it out. I agree that stripping should totally be applied here, but as we've seen, it does expose at least one bug elsewhere.

So: do we leave it out, or brace ourselves for a bunch of issues that may creep up sooner or later? I reckon it's up to Jaap to decide. Either way, changes need to be made in one direction or the other.

Reply:
Hi,
please check if  the change I commited solves the issue you ran into.